all/generate/.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,56 @@
+# This job runs the project tests
+
+name: Build and test on push
+
+# Trigger the workflow on push or pull request to master or develop
+
+on:
+  push:
+    branches: [develop]
+
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Run tests on DEVELOP branch
+        if: ${{ github.ref == 'refs/heads/develop'}}
```

Comment:
why is this just for develop? what happens for other branches (where PRs have not been raised yet)?

is there anyway to not have this duplicated to avoid divergence?

Reply:
Thanks.

**Other branches**

In the Travis configuration, I don't think we ran tests on branch creation. It was only when you made a PR from new branch to `develop`? That said, I am happy to see if we can add branch creation? Though it might lead to multiple test runs: first one when we create a new branch, then another when we create a PR to `develop`

Duplicate 

Unfortunately if we want to have two separate slack messages, then we have to split between two separate jobs. As an alternative, we can put both jobs in the one file? I think that would be better?